observation_portal/requestgroups/serializers.py

```diff
@@ -656,45 +656,45 @@ class Meta:
             'name': {'error_messages': {'blank': 'Please provide a name.'}}
         }
 
-    @transaction.atomic
     def create(self, validated_data):
         request_data = validated_data.pop('requests')
 
-        request_group = RequestGroup.objects.create(**validated_data)
-
-        for r in request_data:
-            configurations_data = r.pop('configurations')
-
-            location_data = r.pop('location', {})
-            windows_data = r.pop('windows', [])
-            request = Request.objects.create(request_group=request_group, **r)
-
-            if validated_data['observation_type'] != RequestGroup.DIRECT:
-                Location.objects.create(request=request, **location_data)
-                for window_data in windows_data:
-                    Window.objects.create(request=request, **window_data)
-
-            for configuration_data in configurations_data:
-                instrument_configs_data = configuration_data.pop('instrument_configs')
-                acquisition_config_data = configuration_data.pop('acquisition_config')
-                guiding_config_data = configuration_data.pop('guiding_config')
-                target_data = configuration_data.pop('target')
-                constraints_data = configuration_data.pop('constraints')
-                configuration = Configuration.objects.create(request=request, **configuration_data)
-
-                AcquisitionConfig.objects.create(configuration=configuration, **acquisition_config_data)
-                GuidingConfig.objects.create(configuration=configuration, **guiding_config_data)
-                Target.objects.create(configuration=configuration, **target_data)
-                Constraints.objects.create(configuration=configuration, **constraints_data)
-
-                for instrument_config_data in instrument_configs_data:
-                    rois_data = []
-                    if 'rois' in instrument_config_data:
-                        rois_data = instrument_config_data.pop('rois')
-                    instrument_config = InstrumentConfig.objects.create(configuration=configuration,
-                                                                        **instrument_config_data)
-                    for roi_data in rois_data:
-                        RegionOfInterest.objects.create(instrument_config=instrument_config, **roi_data)
+        with transaction.atomic():
```

Comment:
for whatever reason, reducing the scope of this transaction.atomic so it didn't include the debit_ipp_time and the cache.set solved one of the deadlocking issues. Specifically, leaving the cache.set within the transaction.atomic didn't work, but I'm not sure why yet

Reply:
Weird. I could see how excluding debit_ipp_time could help, cause that actually touches the db, but setting the cache doesn't... Does it sometimes take a long time to set the cache?

Reply:
no, not that I could see. It was literally the difference between deadlocking forever (or until the 5 minutes worker timeout happens) or not deadlocking. I'll spend some time looking into it a bit more when I've gotten the other things worked out